crates/noirc_frontend/src/hir/def_map/mod.rs

```diff
@@ -77,7 +77,14 @@ impl CrateDefMap {
 
         // First parse the root file.
         let root_file_id = context.crate_graph[crate_id].root_file_id;
-        let ast = parse_file(&mut context.file_manager, root_file_id, errors);
+        let mut ast = parse_file(&mut context.file_manager, root_file_id, errors);
+
+        // TODO(#1850): This check should be removed once we fully move over to the new SSA pass
+        // Compiling with the old SSA pass will lead to duplicate method definitions between
+        // the `slice` and `array` modules of the stdlib
+        if !context.def_interner.enable_slices && crate_id == CrateId::new(1) {
```

Comment:
Which crate is CrateId 1, the stdlib I assume? Can we move it into a constant and assert the stdlib crate actually has this Id when it is used?

Reply:
Yeah it is the stdlib. I will do that

Reply:
This is an error while I was working with one local crate. The stdlib is going to always be the last crate (this PR https://github.com/noir-lang/noir/pull/1840 recently made sure we add the stdlib in `check_crate` rather than the resolver). I updated where we check against the last crate in `collect_defs` and I added a comment. As per the description in PR #1840 it would be ideal to move the stdlib to be the first crate but that requires a larger refactor.